### PR TITLE
DHFPROD-10258: Ingestion performance changes with XQuery MLCP transform

### PR DESCRIPTION
**Pull request by ryanjdew**
#### Checklist: 
```diff
- Note: do not change the below
```

- ##### Owner:

- [x] JIRA_ID included in all the commit messages
- [x] PR title is in the format JIRA_ID:Title
- [x] Rebase the branch with upstream
- [x] Squashed all commits into a single commit
- [x] Added Tests
- [x] Ran newly added/edited cypress tests on Firefox locally
  

- ##### Reviewer:

- [x] Reviewed Tests
- [x] Added to Release Wiki

### CODE REVIEW

**Review by aniamy — APPROVED** [review]

File: marklogic-data-hub-central/src/main/java/com/marklogic/hub/central/controllers/ModelController.java [hunk 1]
```diff
@@ -77,7 +70,7 @@ public ResponseEntity<?> getHubCentralConfig() {
     public ResponseEntity<?> setHubCentralConfig(@RequestBody ObjectNode hubCentralConfig) {
         ObjectNode hubCentralConfigBase = getHubCentralConfigObject();
         String docUri = "/config/hubCentral.json";
-        hcManager.deployHubCentralConfig(getHubClient(), mergeObjects(hubCentralConfigBase, hubCentralConfig), docUri);
+        HubCentralManager.deployHubCentralConfig(getHubClient(), mergeObjects(hubCentralConfigBase, hubCentralConfig), docUri);
```
Comment (aniamy) [inline marklogic-data-hub-central/src/main/java/com/marklogic/hub/central/controllers/ModelController.java:73]: Line 54 should be removed if we are not using hcManager anymore.

File: marklogic-data-hub/src/main/java/com/marklogic/hub/DataHub.java [hunk 2]
```diff
@@ -18,6 +18,7 @@
 
 import com.marklogic.hub.error.ServerValidationException;
 import com.marklogic.hub.flow.FlowRunner;
+
```
Comment (aniamy) [inline marklogic-data-hub/src/main/java/com/marklogic/hub/DataHub.java:21]: extra space here :)

File: marklogic-data-hub/src/main/java/com/marklogic/hub/hubcentral/conversion/FlowConverter.java [hunk 3]
```diff
@@ -241,14 +245,14 @@ else if (StepDefinitionType.MAPPING.equals(stepDefType)) {
                 File stepFile = targetDir.resolve(stepFileName).toFile();
                 logger.info(format("Creating step artifact '%s'", stepFile.toString()));
                 if (stepFile.exists()) {
-                    String msg = "Step artifact '" + stepFile.toString() + "' already exists. The step artifact will be written to ";
+                    String msg = "Step artifact '" + stepFile + "' already exists. The step artifact will be written to ";
```
Comment (aniamy) [inline marklogic-data-hub/src/main/java/com/marklogic/hub/hubcentral/conversion/FlowConverter.java:248]: Why won't we need the toString here? We also do toString in line 246.

File: marklogic-data-hub/src/main/java/com/marklogic/hub/flow/impl/FlowRunnerImpl.java [hunk 4]
```diff
@@ -42,34 +60,34 @@ public class FlowRunnerImpl implements FlowRunner {
     // Only populated when constructed with a HubClient, which implies that a HubProject is not available
     private HubClient hubClient;
 
-    private AtomicBoolean isRunning = new AtomicBoolean(false);
-    private AtomicBoolean isJobCancelled = new AtomicBoolean(false);
-    private AtomicBoolean isJobSuccess = new AtomicBoolean(true);
-    private AtomicBoolean jobStoppedOnError = new AtomicBoolean(false);
+    final AtomicBoolean isRunning = new AtomicBoolean(false);
```
Comment (aniamy) [inline marklogic-data-hub/src/main/java/com/marklogic/hub/flow/impl/FlowRunnerImpl.java:63]: Shouldn't we keep the private and add getters/setters if needed?

File: marklogic-data-hub/src/main/java/com/marklogic/hub/impl/FlowManagerImpl.java [hunk 5]
```diff
@@ -119,7 +133,7 @@ public Flow getLocalFlow(String flowName) {
         JsonNode node = getLocalFlowAsJSON(flowName);
         if(node != null) {
             Flow newFlow = createFlowFromJSON(node);
-            if (newFlow != null && newFlow.getName().length() > 0) {
+            if (newFlow != null && !newFlow.getName().isEmpty()) {
```
Comment (aniamy) [inline marklogic-data-hub/src/main/java/com/marklogic/hub/impl/FlowManagerImpl.java:136]: To check strings you can use apache's StringUtils.isEmpty /isNotEmpty

File: marklogic-data-hub/src/main/java/com/marklogic/hub/step/impl/WriteStepRunner.java [hunk 6]
```diff
@@ -502,8 +506,14 @@ private RunStepResponse runIngester(RunStepResponse runStepResponse, Collection<
 
             runStepResponse.setCounts(stepMetrics.getSuccessfulEventsCount() + stepMetrics.getFailedEventsCount(),stepMetrics.getSuccessfulEventsCount(), stepMetrics.getFailedEventsCount(), stepMetrics.getSuccessfulBatchesCount(), stepMetrics.getFailedBatchesCount());
             runStepResponse.withStatus(stepStatus);
-            if (errorListener.getThrowables().size() > 0) {
-                runStepResponse.withStepOutput(errorListener.getThrowables().stream().map(Throwable::getLocalizedMessage).collect(Collectors.toList()));
+            if (!errorListener.getThrowables().isEmpty()) {
+                runStepResponse.withStepOutput(
+                        errorListener
+                                .getThrowables().stream().filter(Objects::nonNull)
+                                .map((throwable -> (throwable.getMessage() != null) ? throwable.getLocalizedMessage(): throwable.toString()))
```
Comment (aniamy) [inline marklogic-data-hub/src/main/java/com/marklogic/hub/step/impl/WriteStepRunner.java:513]: Suggestion: use toString instead to getLocalizedMessage

**Review by prolonMarkLogic — APPROVED** [review]

**Review by aniamy — APPROVED** [review]

**Review by rahulvudutala — APPROVED** [review]

**Pull request merged**